### PR TITLE
added PATCH req in users route

### LINKED ISSUE #1: Add PATCH method
Currently, the project supports the following operations: `GET`

Add `PATCH` method to the existing `/users` endpoint.

The method should not manipulate the data in the database. It should send the same data that is incoming, with a positive response code.

### PR DESCRIPTION
**Pull request by ritik307**
# Related Issue
- closes #1 

# Proposed Changes
- added `PATCH` route in `users.ts`

# Checklist
- [x] Tests

### CODE REVIEW

**Comment by ritik307** [comment]
@ShauryaAg @paramsiddharth  plz review my PR.

**Review by ShauryaAg — CHANGES_REQUESTED** [review]
@ritik307 Please lint the file, according to the rules defined in config files.

You just need to run `npm run lint`

**Comment by ritik307** [comment]
@ShauryaAg Done! got no errors in the file in which I made changes.

File: src/api/routes/users.ts [hunk 1]
```diff
@@ -38,7 +36,6 @@ router.get("/:userId", (req: Request, res: Response, next: NextFunction) => {
                 });
             }
         })
-        // Single user invalid
```
Comment (ShauryaAg) [inline src/api/routes/users.ts:41]: Why was this line removed?

File: src/api/routes/users.ts [hunk 2]
```diff
@@ -47,11 +44,23 @@ router.get("/:userId", (req: Request, res: Response, next: NextFunction) => {
         });
 });
 
-router.patch("/", (req: Request, res: Response, next: NextFunction) => {
+router.patch("/:id", (req: Request, res: Response, next: NextFunction) => {
     const jsonData=req.body;
-    jsonData["timestamp"]=dateFormat(new Date(), "yyyy-mm-dd HH-MM");
     
-    res.status(200).json(jsonData);
+    // jsonData['timestamp']=dateFormat(new Date(), "yyyy-mm-dd HH-MM")
```
Comment (ShauryaAg) [inline src/api/routes/users.ts:50]: Remove this unnecessary line.

File: src/api/routes/users.ts [hunk 3]
```diff
@@ -47,11 +44,23 @@ router.get("/:userId", (req: Request, res: Response, next: NextFunction) => {
...
+    User.findById(req.params.id).exec().then((user:any)=>{
+        
+        const { first_name=user.first_name,last_name=user.last_name,email=user.email }=jsonData;
```
Comment (ShauryaAg) [inline src/api/routes/users.ts:53]: Put spaces before and after `=` sign

File: src/api/routes/users.ts [hunk 4]
```diff
@@ -47,11 +44,23 @@ router.get("/:userId", (req: Request, res: Response, next: NextFunction) => {
         });
 });
 
-router.patch("/", (req: Request, res: Response, next: NextFunction) => {
+router.patch("/:id", (req: Request, res: Response, next: NextFunction) => {
```
Comment (ShauryaAg) [inline src/api/routes/users.ts:47]: Change `id` to `userId`

**Review by ShauryaAg — CHANGES_REQUESTED** [review]

**Comment by ritik307** [comment]
@ShauryaAg Done! Plz check

**Review by ShauryaAg — APPROVED** [review]

**Comment by ritik307** [comment]
@ShauryaAg @paramsiddharth plz add appropriate GSSOC tag to this isuse.

**Pull request merged**